core/src/replay_stage.rs

```diff
@@ -673,6 +686,57 @@ impl ReplayStage {
         Ok(())
     }
 
+    fn snapshot_tower(tower_snapshot_path: &PathBuf, tower: &Tower) -> Result<()> {
+        let timer = Instant::now();
+        fs::create_dir_all(tower_snapshot_path)?;
+        let mut snapshot_file = File::create(tower_snapshot_path.join(TOWER_SNAPSHOT_NAME))?;
+        snapshot_file.write_all(&bincode::serialize(&tower).expect("tower serialize failed"))?;
```

Comment:
You should create a new file here then do a rename here so that if you crash mid-flush you don't end up with a corrupt file.

Reply:
Consider serialize_into(file) so that the whole Tower doesn't have to be in memory.